### PR TITLE
Feature/sync clean

### PR DESCRIPTION
**Pull request by ianb**
This implements the sync client for the HTML repo and Jetpack addon.  There's still lots of work still to do (https://bugzilla.mozilla.org/buglist.cgi?product=Web%20Apps&component=AppSync&resolution=---&list_id=1748213) but I think it would be better if this was merged and those later features and fixes were small branches.

I've created a separate branch from feature/sync and done a rebase to clean up history a bit, but otherwise this is an equivalent branch.  There's still some rearrangement that happened in the history, and especially `sync/sync.js` is better reviewed in its entirety than as a series of commits (it was previously multiple files).

### CODE REVIEW

File: addons/jetpack/lib/main.js [hunk 1]
```diff
@@ -313,6 +316,49 @@ function setupAboutPageMods() {
...
+    include: APP_SYNC_URL + "/login.html",
+    contentScriptWhen: "end",
+    contentScript: "var button = document.getElementById('signin_button');" +
+      "button.onclick = function() {" +
+      "  unsafeWindow.navigator.wrappedJSObject.id.getVerifiedEmail(function(assertion) {" +
+      "    self.postMessage(assertion);" +
+      "  });" +
+      "};",
+    onAttach: function(worker) {
+      worker.on("message", function(data) {
```
Comment (mixedpuppy) [inline addons/jetpack/lib/main.js]: isn't this going to catch all message events, shouldn't we use worker.port.emit('verifiedEmail', assertion) and worker.port.on('verifiedEmail', fn)?

File: addons/jetpack/lib/main.js [hunk 2]
```diff
@@ -313,6 +316,49 @@ function setupAboutPageMods() {
   });
 }
 
+/**
+ * setupLogin
+ * Shows a jetpack panel to get a BrowserID assertion from the user to
+ * authenticate to the sync service
+ */
+function setupLogin(service) {
+  let wm = Components.classes["@mozilla.org/appshell/window-mediator;1"].getService(Components.interfaces.nsIWindowMediator);
```
Comment (mixedpuppy) [inline addons/jetpack/lib/main.js]: Cc and Ci?

File: addons/jetpack/lib/main.js [hunk 3]
```diff
@@ -313,6 +316,49 @@ function setupAboutPageMods() {
...
+          audience: APP_SYNC_URL 
+        }, function(err, info) {
+          console.log("Got back from login " + JSON.stringify(err) + info);
+          service.syncNow();
+        });
+      });
+    }
+  });
+
+  panel.show(button);
```
Comment (mixedpuppy) [inline addons/jetpack/lib/main.js:360]: show a login panel on startup?  I'll make a UX bug on this.

File: site/jsapi/include.html [hunk 4]
```diff
@@ -1,4 +1,4 @@
-<!DOCTYPE html>
+>><!DOCTYPE html>
```
Comment (mixedpuppy) [inline site/jsapi/include.html]: why the >> before doctype?

Comment (ianb) [inline site/jsapi/include.html]: Fixed

File: addons/jetpack/lib/main.js (hunk 3 above)
Comment (anantn) [inline addons/jetpack/lib/main.js:360]: https://bugzilla.mozilla.org/show_bug.cgi?id=704392

File: addons/jetpack/lib/main.js (hunk 1 above)
Comment (anantn) [inline addons/jetpack/lib/main.js]: Fixed.

File: addons/jetpack/lib/main.js (hunk 2 above)
Comment (anantn) [inline addons/jetpack/lib/main.js]: Fixed.

**Pull request merged**